src/iop/channelmixer.c

```diff
@@ -150,9 +187,20 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, void *
                                      ? TRUE
                                      : FALSE;
   const int ch = piece->colors;
+  // Compute weights
```

Comment:
Oh, wrong approah.
You should put ```float weights[CHANNEL_SIZE]``` into ``dt_iop_channelmixer_data_t``, and compute it in ```commit_params()```
And then probably remove ``data->luminosity`` from ``dt_iop_channelmixer_data_t``

(that way you will have code to compute it in only one function, not in two like now)